pkg/kubectl/apply/error.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apply
+
+import (
+	"fmt"
+)
+
+const (
+	// CONFLICT is type identity for conflicts occurs in recorded and remote
+	CONFLICT = "Conflict"
+)
+
+// ConflictError for conflicts detection, Message is for error message.
+// TODO : Detail field is for more details in error, like full path where conflicts happened.
+type ConflictError struct {
+	Type    string
+	Message string `json:"message"`
```

Comment:
Are json tags useful for this type?  If so, `Type` should have a tag if that field is needed (unless Type can be removed of course).

Reply:
Removed Type field.